multistream-2.0/spec.md

```diff
@@ -0,0 +1,319 @@
+# Multistream 2.0
+
+This proposal describes a replacement protocol for multistream-select.
+
+## Protocols
+
+This document proposes 5 new, micro-protocols with two guiding principles:
+
+1. Composition over complexity.
+2. Every byte and round-trip counts.
+
+This document *does not*, in fact, propose a protocol *negotiation* protocol.
+Instead, it proposes a set of stream/protocol management protocols that can be
+composed to flexibly negotiate protocols.
+
+First, this document proposes 4 protocol "negotiation" protocols. "Negotiation"
+is in quotes because none of these protocols actually involve negotiating
+anything.
+
+1. `multistream/advertise`: Inform the remote end about which protocols we
+   speak. This should partially replace the current identify protocol.
+2. `multistream/use`: Selects the stream's protocol using a multicodec.
```

Comment:
I'm not a fan of these names. How about `multistream/use-muilticodec`, `multistream/use-dynamic`, `multistream/use-contextual`?

Reply:
Renamed use -> multicodec, dynamic -> string, contextual -> dynamic.